src/vm/runners/builtin_runner/mod.rs

```diff
@@ -257,108 +258,102 @@ impl BuiltinRunner {
         }
     }
 
+    fn cells_per_instance(&self) -> u32 {
+        match self {
+            BuiltinRunner::Bitwise(builtin) => builtin.cells_per_instance,
+            BuiltinRunner::EcOp(builtin) => builtin.cells_per_instance,
+            BuiltinRunner::Hash(builtin) => builtin.cells_per_instance,
+            BuiltinRunner::RangeCheck(builtin) => builtin.cells_per_instance,
+            BuiltinRunner::Output(_) => 0,
+            BuiltinRunner::Keccak(builtin) => builtin.cells_per_instance,
+            BuiltinRunner::Signature(builtin) => builtin.cells_per_instance,
+        }
+    }
+
+    fn n_input_cells(&self) -> u32 {
+        match self {
+            BuiltinRunner::Bitwise(builtin) => builtin.n_input_cells,
+            BuiltinRunner::EcOp(builtin) => builtin.n_input_cells,
+            BuiltinRunner::Hash(builtin) => builtin.n_input_cells,
+            BuiltinRunner::RangeCheck(builtin) => builtin.n_input_cells,
+            BuiltinRunner::Output(_) => 0,
+            BuiltinRunner::Keccak(builtin) => builtin.n_input_cells,
+            BuiltinRunner::Signature(builtin) => builtin.n_input_cells,
+        }
+    }
+
+    pub fn name(&self) -> &'static str {
+        match self {
+            BuiltinRunner::Bitwise(_) => "bitwise",
+            BuiltinRunner::EcOp(_) => "ec_op",
+            BuiltinRunner::Hash(_) => "hash",
+            BuiltinRunner::RangeCheck(_) => "range_check",
+            BuiltinRunner::Output(_) => "output",
+            BuiltinRunner::Keccak(_) => "keccak",
+            BuiltinRunner::Signature(_) => "ecdsa",
+        }
+    }
+
     pub fn run_security_checks(&self, vm: &mut VirtualMachine) -> Result<(), VirtualMachineError> {
         if let BuiltinRunner::Output(_) = self {
             return Ok(());
         }
-
-        let (cells_per_instance, n_input_cells) = match self {
-            BuiltinRunner::Bitwise(x) => (x.cells_per_instance, x.n_input_cells),
-            BuiltinRunner::EcOp(x) => (x.cells_per_instance, x.n_input_cells),
-            BuiltinRunner::Hash(x) => (x.cells_per_instance, x.n_input_cells),
-            BuiltinRunner::RangeCheck(x) => (x.cells_per_instance, x.n_input_cells),
-            BuiltinRunner::Output(_) => unreachable!(),
-            BuiltinRunner::Keccak(x) => (x.cells_per_instance, x.n_input_cells),
-            BuiltinRunner::Signature(ref x) => (x.cells_per_instance, x.n_input_cells),
+        let cells_per_instance = self.cells_per_instance() as usize;
+        let n_input_cells = self.n_input_cells() as usize;
+        let builtin_segment_index = self
+            .base()
+            .to_usize()
+            .ok_or(VirtualMachineError::NegBuiltinBase)?;
+        // If the builtin's segment is empty, there are no security checks to run
+        let builtin_segment = match vm.memory.data.get(builtin_segment_index) {
+            Some(segment) => segment,
+            None => return Ok(()),
```

Comment:
Is `segment.is_empty()` a possibility? If it is then it should also `Return Ok(())`, right?
```suggestion
        // If the builtin's segment is empty, there are no security checks to run
        let builtin_segment = match vm.memory.data.get(builtin_segment_index) {
            Some(segment) if !segment.is_empty() => segment,
            _ => return Ok(()),
```

Reply:
Good call